### ISSUE TITLE
GraphQL Types clashing 

### ISSUE THREAD

**Issue by dsilvam** [dotCMS/core#19660]
```graphql.AssertException: All types within a GraphQL schema must have unique names. No two provided types may have the same name.
No provided type may have a name which conflicts with any built in types (including Scalar and Introspection types).
You have redefined the type 'Body' from being a 'GraphQLObjectType' to a 'GraphQLObjectType'
	at graphql.schema.GraphQLTypeCollectingVisitor.assertTypeUniqueness(GraphQLTypeCollectingVisitor.java:105) ~[graphql-java-13.0.jar:?]
	at graphql.schema.GraphQLTypeCollectingVisitor.visitGraphQLObjectType(GraphQLTypeCollectingVisitor.java:38) ~[graphql-java-13.0.jar:?]
	at graphql.schema.GraphQLObjectType.accept(GraphQLObjectType.java:174) ~[graphql-java-13.0.jar:?]
	at graphql.schema.TypeTraverser$TraverserDelegateVisitor.enter(TypeTraverser.java:71) ~[graphql-java-13.0.jar:?]
	at graphql.util.Traverser.traverse(Traverser.java:147) ~[graphql-java-13.0.jar:?]
	at graphql.schema.TypeTraverser.doTraverse(TypeTraverser.java:58) ~[graphql-java-13.0.jar:?]
	at graphql.schema.TypeTraverser.depthFirst(TypeTraverser.java:50) ~[graphql-java-13.0.jar:?]
	at graphql.schema.TypeTraverser.depthFirst(TypeTraverser.java:38) ~[graphql-java-13.0.jar:?]
	at graphql.schema.SchemaUtil.allTypes(SchemaUtil.java:42) ~[graphql-java-13.0.jar:?]
	at graphql.schema.GraphQLSchema.<init>(GraphQLSchema.java:102) ~[graphql-java-13.0.jar:?]
	at graphql.schema.GraphQLSchema.<init>(GraphQLSchema.java:37) ~[graphql-java-13.0.jar:?]
	at graphql.schema.GraphQLSchema$Builder.build(GraphQLSchema.java:411) ~[graphql-java-13.0.jar:?]
	at com.dotcms.graphql.business.GraphqlAPIImpl.generateSchema(GraphqlAPIImpl.java:138) ~[classes/:?]
	at com.dotcms.graphql.business.GraphqlAPIImpl.getSchema(GraphqlAPIImpl.java:75) ~[classes/:?]
	at com.dotcms.graphql.DotGraphQLSchemaProvider.getSchema(DotGraphQLSchemaProvider.java:27) ~[classes/:?]
```

**Comment by dsilvam** [comment]
PR: https://github.com/dotCMS/core/pull/19662

**Comment by dsilvam** [comment]
Note to QA:

1. import custom dataset (ask me for it) 
2. run a full reindex
3. test querying a graphql collection (card collection for example)

**Comment by bryanboza** [comment]
Fixed, tested with the provided dataset and works as expected

**Comment by john-thomas-dotcms** [comment]
@jcastro-dotcms is there a reason we can't include this fix in an LTS patch release?

**Comment by jcastro-dotcms** [comment]
@john-thomas-dotcms 

TL;DR There were too many code changes that needed to be ported to the LTS patch, even code that was not part of the PR itself.

The reason why this fix didn't make it to the LTS patch was because a huge amount of code changed between the dotCMS version that the code fix was created against, and the LTS versions. Also, this PR relied on code changes that were not in the LTS patches, and were not part of the PR either. The number of changes started to grow, and the specific changes required updating very sensitive pieces of dotCMS as well, which is extremely risky.
